src/libopenrave/libopenrave.cpp

```diff
@@ -1271,22 +1272,22 @@ InterfaceBasePtr RaveCreateInterface(EnvironmentBasePtr penv, InterfaceType type
 
 RobotBasePtr RaveCreateRobot(EnvironmentBasePtr penv, const std::string& name)
 {
-    return RaveGlobal::instance()->GetDatabase()->CreateRobot(penv,name);
+    return RaveInterfaceCast<RobotBase>(RaveCreateInterface(penv, PT_Robot, name));
```

Comment:
@undisputed-seraphim why not use `RaveGlobal::instance()->GetDatabase()->Create` directly?

Reply:
Sure I can, but there's no difference between those two ways, the compiler will inline all of them during optimization.

I went with this way because it's fewer characters for the human programmer to parse, and the defintion of `RaveCreateInterface` is right above.

Reply:
actually there is, please use `RaveGlobal::instance()->GetDatabase()-`